src/supremm/summarize.py

```diff
@@ -1,54 +1,27 @@
-#!/usr/bin/env python3
-""" Summarize module """
+""" Definition of the summarize API """
+from abc import ABC, abstractmethod
 
-import datetime
-
-from ctypes import c_uint
-from pcp import pmapi
-import cpmapi as c_pmapi
-import time
-import logging
-import traceback
-from supremm.plugin import NodeMetadata
-from supremm.rangechange import RangeChange, DataCache
-from supremm.pcpcinterface import pcpcinterface
-
-import numpy
-import copy
-
-VERSION = "1.0.6"
+VERSION = "2.0.0"
 TIMESERIES_VERSION = 4
 
 
-class ArchiveMeta(NodeMetadata):
-    """ container for achive metadata """
-    def __init__(self, nodename, nodeidx, archivedata):
-        self._nodename = nodename
-        self._nodeidx = nodeidx
-        self._archivedata = archivedata
-
-    nodename = property(lambda self: self._nodename)
-    nodeindex = property(lambda self: self._nodeidx)
-    archive = property(lambda self: self._archivedata)
-
-class Summarize(object):
-    """
-    Summarize class is responsible for iteracting with the pmapi python code
-    and managing the calls to the various analytics to process the data
+class Summarize(ABC):
+    """ Abstract base class describing the job summarization interface.
+        Currently only interfaces with PCP archives and is subject to change.
```

Comment:
Please double check this comment is correct. Why create an abstract base class with only one implementtaion?

Reply:
Resolved in https://github.com/ubccr/supremm/pull/259/commits/a34241f9050f4e6fcc5b9fb084199663ce3c9fe2